server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -5341,6 +5341,13 @@ protected void updateVncPasswordIfItHasChanged(String originalVncPassword, Strin
     public void finalizeExpunge(VirtualMachine vm) {
     }
 
+    private void checkForceStopVmPermission(Account callingAccount) {
+        if (!AllowUserForceStopVm.valueIn(callingAccount.getId())) {
+            logger.error(String.format("Parameter [%s] can only be passed by Admin accounts or when the allow.user.force.stop.vm config is true for the account.", ApiConstants.FORCED));
```

Comment:
Just a minor suggestion regarding the `log4j2` syntax.

```suggestion
            logger.error("Parameter [{}] can only be passed by Admin accounts or when the allow.user.force.stop.vm config is true for the account.", ApiConstants.FORCED);
```